test/forge/base/BaseTest.t.sol

```diff
@@ -0,0 +1,455 @@
+            IAgglayerGER(Constants.GER_MANAGER_ADDRESS),
+            IERC20Upgradeable(Constants.POL_TOKEN_ADDRESS),
+            IAgglayerBridge(Constants.BRIDGE_ADDRESS),
+            AgglayerManager(Constants.ROLLUP_MANAGER_ADDRESS),
+            IAgglayerGateway(Constants.AGGLAYER_GATEWAY_ADDRESS)
+        );
+        return aggchainFEPImpl;
+    }
+
+    /**
+     * @dev Deploy AggchainECDSAMultisig implementation contract
+     * @return The deployed AggchainECDSAMultisig contract
+     */
+    function deployAggchainECDSA() public returns (AggchainECDSAMultisig) {
+        // === AGGCHAINECDSAMULTISIG IMPLEMENTATION DEPLOYMENT ===
+        vm.prank(owner);
+        aggchainECDSAImpl = new AggchainECDSAMultisig(
+            IAgglayerGER(Constants.GER_MANAGER_ADDRESS),
+            IERC20Upgradeable(Constants.POL_TOKEN_ADDRESS),
+            IAgglayerBridge(Constants.BRIDGE_ADDRESS),
+            AgglayerManager(Constants.ROLLUP_MANAGER_ADDRESS),
+            IAgglayerGateway(Constants.AGGLAYER_GATEWAY_ADDRESS)
+        );
+        return aggchainECDSAImpl;
+    }
+
+    /**
+     * @dev Etch PolygonZkEVMDeployer bytecode from Hardhat artifacts
+     * @param _owner The owner address for the deployer
+     * @return deployerAddress The address where the deployer was etched
+     */
+    function _etchPolygonZkEVMDeployer(address _owner) internal returns (address) {
+        // Read the deployedBytecode from Hardhat artifacts
+        string memory artifact = vm.readFile(POLYGON_ZKEVM_DEPLOYER_ARTIFACT_PATH);
+        bytes memory deployedBytecode = vm.parseJsonBytes(artifact, ".deployedBytecode");
+
+        // Generate a deterministic address for the deployer
+        address deployerAddress = makeAddr("PolygonZkEVMDeployer");
+
+        // Etch the bytecode at the address
+        vm.etch(deployerAddress, deployedBytecode);
+
+        // Set the owner storage slot (slot 0 in Ownable)
+        vm.store(deployerAddress, bytes32(uint256(0)), bytes32(uint256(uint160(_owner))));
+
+        return deployerAddress;
+    }
+
+    /**
+     * @dev Etch AgglayerTimelock bytecode from Hardhat artifacts with immutable replacement
+     * @param _minDelay Minimum delay for timelock operations
+     * @param _proposers Array of proposer addresses
+     * @param _executors Array of executor addresses
+     * @param _agglayerManagerAddr The agglayer manager contract (used for emergency state)
+     * @return timelockAddress The address where the timelock was etched
+     */
+    function _etchAgglayerTimelock(
+        uint256 _minDelay,
+        address[] memory _proposers,
+        address[] memory _executors,
+        address _agglayerManagerAddr
+    ) internal returns (address) {
+        // Read the deployedBytecode from Hardhat artifacts
+        string memory artifact = vm.readFile(AGGLAYER_TIMELOCK_ARTIFACT_PATH);
+        bytes memory deployedBytecode = vm.parseJsonBytes(artifact, ".deployedBytecode");
```

Comment:
why do we reuse the bytecode generated by hardhat?
Why we don't deploy with froundary?¿
also replaceImmutable, if im not mistaken  it will replace all btyes32(0) as well... which will have a lot of known and unknown consecuences